### PR TITLE
Upgrade EFA to version 1.8.4

### PR DESCRIPTION
**Pull request by tilne**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,22 @@ aws-parallelcluster-cookbook CHANGELOG
...
+-----
+
+**ENHANCEMENTS**
+
+**CHANGES**
+- Upgrade EFA installer to version 1.8.4
+  - Kernel module: efa-1.5.1 (no change)
+  - RDMA core: rdma-core-25.0 (distributed only) (no change)
+  - Libfabric: libfabric-aws-1.9.0amzn1.1 (no change)
+  - Open MPI: openmpi40-aws-4.0.3 (vs. openmpi40-aws-4.0.2)
```
Comment (enrico-usai) [inline CHANGELOG.md:16]: minor: let's use the same convention we have in previous changes: (updated from .. in place of "vs.")

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,22 @@ aws-parallelcluster-cookbook CHANGELOG
...
+  - EFA profile: efa-profile-1.0 (new)
+  - EFA config: efa-config-1.0 (new)
```
Comment (enrico-usai) [inline CHANGELOG.md:18]: do you know what are efa-profile and efa-config for?
I'd add an explanation for these two new files in the commit message, to remember us in the future why EFA added these new files and how are we using them.

Comment (tilne) [reply CHANGELOG.md:18]: No. I just copied it from the email that subspace sent out to be honest.

File: CHANGELOG.md (hunk 1 above)
Comment (tilne) [reply CHANGELOG.md]: done

**Pull request merged**